pkg/apis/apm/v1/webhook_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v1_test
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	apmv1 "github.com/elastic/cloud-on-k8s/pkg/apis/apm/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/test"
+	"github.com/stretchr/testify/require"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestWebhook(t *testing.T) {
+	testCases := []test.ValidationWebhookTestCase{
+		{
+			Name:      "create-valid",
+			Operation: admissionv1beta1.Create,
+			Object: func(t *testing.T, uid string) []byte {
+				apm := mkApmServer(uid)
+				return serialize(t, apm)
```

Comment:
Does it maybe make sense to offload the serialization calls to RunValidationWebhookTests, and change the ValidationWebhookTestCase.Object fields to a generic object? As it is there's a lot of boiler plate in TestCase.Object. That said, even if it does make sense I'm still okay with merging this in today as is and leaving that for a future refactoring.